### PR TITLE
{tools}[system/system] zsh v5.9

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/jfgrimm/b68cecc6ce7a605dbc508943503a4582 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/88ea1ac6de410dc6e319e1982d8ef4e3 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Failed during parsing of the easyconfigs, so no ecs were built (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux Ubuntu 20.04.2 LTS (Focal Fossa), x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.8.5
See https://gist.github.com/branfosj/836966a248c359a75ec8ecae3238edc9 for a full test report.

File: easybuild/easyconfigs/z/zsh/zsh-5.9.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+configopts = '--with-tcsetpgrp'  # needed to build in non-login shells
+
+modextrapaths = {'FPATH': 'share/zsh/%(version)s/functions'}
+
+sanity_check_paths = {
+    'files': ['bin/zsh'],
+    'dirs': ['lib/zsh/%(version)s', 'share'],
+}
+
```
Comment (branfosj) [inline easybuild/easyconfigs/z/zsh/zsh-5.9.eb:27]: Should we add `zsh --version` as a `sanity_check_cmd`?

**Review by branfosj — DISMISSED** [review]

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @jfgrimm!

**Pull request merged**